extensions/vscode/src/views/homeView.ts

```diff
@@ -944,7 +945,16 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
     }
   }
 
-  private inputSecretName = async () => {
+  private inputSecretName = async (
+    environment: EnvironmentConfig | undefined,
+  ) => {
+    const existingKeys = new Set();
+    if (environment) {
+      for (const secret in environment) {
+        existingKeys.add(secret);
+      }
+    }
+
```

Comment:
I'd suggest you create this set each time you are called to validate. That will allow a user to encounter the error, then correct it by removing the old secret, and then continue forward with the new secret definition.

I'd also suggest you perform this validation during the `finalValidation` callback rather than the one which is called on every keystroke. That should address any concerns raised with my suggestion above.

Reply:
> create this set each time you are called to validate

That won't quite do it. I got this working and I have to re-get the configuration entirely since it can be lost, have an error, or an entirely different `environment` since the input is using `ignoreFocusOut: true`. You can go into the configuration file and make any changes you want.

Like I said I got this working, but it doesn't feel like a good user or coding experience. There are way more checks in the code, and more avenues for user mistakes.

This feels like a really good reason to only use `ignoreFocusOut` when necessary. I'll add something to our team discussions to go over this.

> this validation during the finalValidation callback

There isn't a way to use our `finalValidation` without creating a `MultiStepInput`. That seems like a good follow-up to create that and use it here.

I do worry about the maintenance cost of that as oppose to using the `validateInput` style of VS Code, but do agree its a better UX.

Reply:
Posting the code here to compare.

```typescript
    return await window.showInputBox({
      title: "Add a Secret",
      prompt: "Enter the name of the secret.",
      ignoreFocusOut: true,
      validateInput: async (value: string) => {
        const activeConfig = await this.state.getSelectedConfiguration();

        if (activeConfig === undefined) {
          return "The selected Configuration has been lost. Please select a Configuration and try again.";
        }
        if (isConfigurationError(activeConfig)) {
          return "The selected Configuration has an error. Please fix the error and try again.";
        }

        const environment = activeConfig.configuration.environment;
        const existingKeys = new Set();
        if (environment) {
          for (const secret in environment) {
            existingKeys.add(secret);
          }
        }

        if (value.length === 0) {
          return "Secret names cannot be empty.";
        }
        if (existingKeys.has(value)) {
          return "There is already an environment variable with this name. Secrets and environment variable names must be unique.";
        }
        return;
      },
    });
```

Reply:
Yeah, ok. I see your point. Thanks for exploring it.

Reply:
I think there is a discussion to be had here, but we should leave it for after `1.2.0`